application/controllers/AdminController.php

```diff
@@ -120,6 +120,30 @@ public function error($message, $sURL = array())
         Yii::app()->end();
     }
 
+    public function actionRemoveTemplateToken()
+    {
+        if (isset(Yii::app()->session['templatetoken'])) {
+            unset(Yii::app()->session['templatetoken']);
+        }
+    }
+
+    public function actionInstallTemplateByToken()
+    {
+        if (isset(Yii::app()->session['templatetoken'])) {
+            Yii::import('application.helpers.admin.token_helper', true);
+            $filename = decodeFilename(Yii::app()->session['templatetoken']);
+            $this->actionRemoveTemplateToken();
+            if (!preg_match('/^[a-zA-Z0-9_\.]*$/', $filename)) {
+                echo "badly formatted file";
+            } else {
+                exec("php application/commands/console.php importsurvey import-file {$filename}", $output);
```

Comment:
I'm not sure this will be needed, since we'll use our message system and cronjobs to distribute and install template files. Correct me if I'm wrong, maybe there's a use-case I'm not aware of?

Reply:
It's gonna be hard for me to approve a PR in which the functionality is spread out like this in different controllers, so starting work with the message queue system might be closer at hand for now.

Reply:
I have no knowledge about the message queue system yet, so I will need to get through a learning curve in order to become able to apply this idea